onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -533,12 +546,21 @@
 
   auto feature_level = D3D_FEATURE_LEVEL_11_0;
   if (IsNPU(adapter.Get())) {
-    feature_level = D3D_FEATURE_LEVEL_1_0_CORE;
+    feature_level = D3D_FEATURE_LEVEL_1_0_GENERIC;
   }
 
   // Create D3D12 Device from DXCore Adapter
   ComPtr<ID3D12Device> d3d12_device;
-  ORT_THROW_IF_FAILED(D3D12CreateDevice(adapter.Get(), feature_level, IID_GRAPHICS_PPV_ARGS(d3d12_device.ReleaseAndGetAddressOf())));
+  if (feature_level == D3D_FEATURE_LEVEL_1_0_GENERIC) {
+      // Attempt to create a D3D_FEATURE_LEVEL_1_0_CORE device first, in case the device supports this
+      // feature level and the D3D runtime does not support D3D_FEATURE_LEVEL_1_0_GENERIC
```

Comment:
Shouldnt CORE always have generic ml? except for meteor lake?

in which case we can special case meteor lake?

Reply:
Core implies generic, but the intent here is working with older runtimes that don't understand generic.